app/common/util/src/main/java/io/syndesis/common/util/Labels.java

```diff
@@ -40,11 +39,11 @@ private Labels() {
      * @return      The sanitized string.
      */
     public static String sanitize(String name) {
-        return trim(name)
+        return trim(name
             .replaceAll(SPACE, DASH)
-            .toLowerCase(Locale.US)
             .chars()
             .filter(i -> !String.valueOf(i).matches(INVALID_CHARACTER_REGEX))
```

Comment:
`.` and `_` are both valid, too. Actually for better understanding I wouldnt use a double negation but sth like `i -> String.valueOf(i).matches(VALID_CHARACTER_REGEX)`. 

Using a regexp for every char is a quite expensive operation, btw, it might be better filter out with 

```
(c >= 'a' && c <= 'z') || 
(c >= '0' && c <= '9') || 
(c >= 'A' && c <= 'Z') || 
c == '.' || c == '-' || c == '_'
```